src/app/clients/ClientsPage.tsx

```diff
@@ -43,13 +45,45 @@ const fullNameSearch = (
     return query.ilike("full_name", `%${state}%`);
 };
 
+const postcodeSearch = (
+    query: PostgrestFilterBuilder<Database["public"], any, any>,
+    state: string
+): PostgrestFilterBuilder<Database["public"], any, any> => {
+    if (state === "") {
+        return query;
+    }
+    if (nullPostcodeDisplay.toLowerCase().includes(state.toLowerCase())) {
+        return query.or(`address_postcode.ilike.%${state}%, address_postcode.is.null`);
+    }
+    return query.ilike("address_postcode", `%${state}%`);
```

Comment:
It looks like it's exactly the same as the parcel one except the database field name (?), in which case I think it might be good to pull it out into a separate helper function somewhere and pass the field name as a parameter (the `common` folder looks like a good place for it maybe?)

Reply:
This makes a lot of sense - I was trying this today and had to deal with some merge conflicts from some refactoring in the dev branch. I think this might be more complicated now that the functions in `filters.ts` under clientsTable and parcelsTable have different types. I need to look into types and generics - might take me some time but I'm happy to do it! Let me know if you want me to pick this up on Monday! :)

Reply:
Ignore the above message - I have just done this. Ready for review!